util/deephash/deephash.go

```diff
@@ -175,48 +183,33 @@ func (h *hasher) print(v reflect.Value) (acyclic bool) {
 				// to allocate, so it's not a win.
 				n := reflect.Copy(reflect.ValueOf(&h.scratch).Elem(), v)
 				w.Write(h.scratch[:n])
-				return true
+				return
 			}
 			fmt.Fprintf(w, "%s", v.Interface())
-			return true
+			return
 		}
-		acyclic = true
 		for i := 0; i < vLen; i++ {
+			// TODO(dsnet): Perform cycle detection for slices,
+			// which is functionally a list of pointers.
+			// See https://github.com/google/go-cmp/blob/402949e8139bb890c71a707b6faf6dd05c92f4e5/cmp/compare.go#L438-L450
 			h.int(i)
-			if !h.print(v.Index(i)) {
-				acyclic = false
-			}
+			h.print(v.Index(i))
 		}
-		return acyclic
 	case reflect.Interface:
-		return h.print(v.Elem())
+		h.print(v.Elem())
 	case reflect.Map:
```

Comment:
Check for nil and write `0` in that case, to mirror the pointer code above?

Reply:
Future update to code :)